### PR TITLE
Force GuiceServiceLocatorGenerator to be the only ServiceLocatorGenerator

### PR DESCRIPTION
**Pull request by FranOis**
Even if it wasn't initialized with install, it should create a GuiceServiceLocatorGenerator, which solves the race condition mentioned in the README.md.

Also made it optional to bind things on to GuiceServlet scopes so that even if you are not using GuiceServlets and only want to link Guice to HK2 it still works.

### CODE REVIEW

**Comment by AlexITC** [comment]
@rkapsi have you ever checked this pull request?

File: README.md [hunk 1]
```diff
@@ -57,6 +57,10 @@ public static void main(String[] args) {
 
 ```
 
+#### META-INF
+
+To ensure Jersey/HK2 SPIs will use the proper ServiceLocationGenerator, simply copy the META-INF folder into your root project's resources folder.
```
Comment (rkapsi) [inline README.md:62]: Doing this is fair enough but how is it different from adding `jersey2-guice-spi` to the classpath. Both are subject to classpath ordering.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rkapsi) [inline pom.xml:1]: This pom file contains some (possibly) private information.

**Comment by jhaber** [comment]
Any chance of getting a release with these changes soon? The change I'm mainly looking for is [here](https://github.com/Squarespace/jersey2-guice/pull/39/files#diff-a1f0825aeeb627cc56eb829c34394860R50). Without this change, an `IllegalStateException` is thrown any time a Jersey class is used before a jersey2-guice app is started up. This includes Jersey client, URI builder, and lots of other classes that don't need a running jersey2-guice app to work. This mainly causes issues with tests and has been reported by a bunch of people [here](https://github.com/HubSpot/dropwizard-guice/issues/95)

**Pull request closed without merging**